File: android/src/main/java/expo/modules/gaodemap/overlays/PolylineViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Polyline 视图 Module
 */
class PolylineViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("PolylineView")

    View(PolylineView::class) {
      Events("onPress")
      
      Prop<List<Map<String, Double>>>("points") { view: PolylineView, points ->
        view.setPoints(points)
      }
      
      Prop<Float>("strokeWidth") { view: PolylineView, width ->
        view.setStrokeWidth(width)
      }
      
      Prop<Int>("strokeColor") { view: PolylineView, color ->
        view.setStrokeColor(color)
      }
      
      Prop<String?>("texture") { view: PolylineView, texture ->
        view.setTexture(texture)
      }
      
      Prop<Boolean>("dotted") { view: PolylineView, dotted ->
        view.setDotted(dotted)
      }
      
      Prop<Boolean>("geodesic") { view: PolylineView, geodesic ->
        view.setGeodesic(geodesic)
      }
      
      Prop<Float>("zIndex") { view: PolylineView, zIndex ->
        view.setZIndex(zIndex)
      }
    }
  }
}

File: ios/overlays/PolylineViewModule.swift
import ExpoModulesCore

public class PolylineViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("PolylineView")
        
        View(PolylineView.self) {
            Events("onPress")
            
            Prop("points") { (view: PolylineView, points: [[String: Double]]) in
                view.setPoints(points)
            }
            
            Prop("width") { (view: PolylineView, width: Double) in
                view.setStrokeWidth(Float(width))
            }
            
            Prop("strokeWidth") { (view: PolylineView, width: Double) in
                view.setStrokeWidth(Float(width))
            }
            
            Prop("color") { (view: PolylineView, color: String) in
                view.setStrokeColor(color)
            }
            
            Prop("strokeColor") { (view: PolylineView, color: String) in
                view.setStrokeColor(color)
            }
            
            Prop("texture") { (view: PolylineView, url: String?) in
                view.setTexture(url)
            }
            
            Prop("dotted") { (view: PolylineView, dotted: Bool) in
                view.setDotted(dotted)
            }
        }
    }
}